MekHQ/src/mekhq/gui/dialog/AutoResolveBehaviorSettingsDialog.java

```diff
@@ -0,0 +1,560 @@
+package mekhq.gui.dialog;
+
+import megamek.client.bot.princess.BehaviorSettings;
+import megamek.client.bot.princess.BehaviorSettingsFactory;
+import megamek.client.bot.princess.CardinalEdge;
+import megamek.client.bot.princess.PrincessException;
+import megamek.client.ui.Messages;
+import megamek.client.ui.baseComponents.MMComboBox;
+import megamek.client.ui.dialogs.helpDialogs.PrincessHelpDialog;
+import megamek.client.ui.swing.MMToggleButton;
+import megamek.client.ui.swing.util.ScalingPopup;
+import megamek.client.ui.swing.util.UIUtil;
+import megamek.logging.MMLogger;
+import mekhq.campaign.Campaign;
+import mekhq.gui.baseComponents.AbstractMHQDialog;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.TitledBorder;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import java.awt.*;
+import java.awt.event.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class AutoResolveBehaviorSettingsDialog
```

Comment:
So, with the amount of code here that _seems_ to be duplicated in the MegaMek bot behavior dialog, I do wonder whether we can reduce code redundancy by extending that. Then you'd only need to add any new functionality, or `@Override` methods you need to change.

Reply:
I will try, but at first I decided against that because the Bot behavior settings dialog had too much to do with changing how the bots work, sending data directly to a scenario and bot forces, it expects the clientGUI to be available, the current scenario, access to the forces, etc.

Reply:
Gotcha. In that case it's probably not worth doing.